File: app/src/main/java/ownmanager/in/fragmentruntimesample/fragments/SingleFragmentMultipleInstance.java
package ownmanager.in.fragmentruntimesample.fragments;


import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;

import ownmanager.in.fragmentruntimesample.R;


/**
 * A simple {@link Fragment} subclass.
 * Use the {@link SingleFragmentMultipleInstance#newInstance} factory method to
 * create an instance of this fragment.
 */
public class SingleFragmentMultipleInstance extends Fragment {

    // Keys for Bundle
    private static final String KEY_1 = "param1";
    private static final String KEY_2 = "param2";


    private String mParam1;
    private String mParam2;


    public SingleFragmentMultipleInstance() {
        // Required empty public constructor
    }

    // Call this to get the new instances - >  SingleFragmentMultipleInstance.newInstance("1","2");

    //Creates new instance
    public static SingleFragmentMultipleInstance newInstance(String param1, String param2) {
        SingleFragmentMultipleInstance fragment = new SingleFragmentMultipleInstance();
        Bundle args = new Bundle();
        args.putString(KEY_1, param1);
        args.putString(KEY_2, param2);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mParam1 = getArguments().getString(KEY_1);
            mParam2 = getArguments().getString(KEY_2);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_single_fragment_multiple_instance, container, false);
    }

}

File: app/src/main/java/ownmanager/in/fragmentruntimesample/fragments/SingleFragmentMultipleInstance.kt
package ownmanager.`in`.fragmentruntimesample.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import ownmanager.`in`.fragmentruntimesample.databinding.FragmentSingleFragmentMultipleInstanceBinding

/**
 * A simple [Fragment] subclass.
 * Use the [SingleFragmentMultipleInstance.newInstance] factory method to
 * create an instance of this fragment.
 */

private const val ARG_PARAM1 = "param1"
private const val ARG_PARAM2 = "param2"

class SingleFragmentMultipleInstance : Fragment() {

    private var _binding: FragmentSingleFragmentMultipleInstanceBinding? = null
    private val binding get() = _binding!!

    private var param1: String? = null
    private var param2: Int? = null


    companion object {
        //Creates new instance
        fun newInstance(param1: String, param2: Int) =
            SingleFragmentMultipleInstance().apply {
                arguments = Bundle().apply {
                    putString(ARG_PARAM1, param1)
                    putInt(ARG_PARAM2, param2)
                }
            }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        arguments?.let {
            param1 = it.getString(ARG_PARAM1)
            param2 = it.getInt(ARG_PARAM2)
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        _binding = FragmentSingleFragmentMultipleInstanceBinding
            .inflate(inflater, container, false)
        bindValues()
        return binding.root
    }

    private fun bindValues() {
        with(binding) {
            instanceCount.text = param2.toString()
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}